### DISCUSSION TITLE
SELinux is preventing /usr/bin/fwupdmgr from map access on the file /root/.cache/dconf/user.

### DISCUSSION THREAD

**Discussion by Tronde** [fwupd/fwupd#5962] [General]
Hi there,  
I'm receiving the message in the title on my RHEL 9.2 system and would like to ask: "Does `fwupdmgr` need map access to the mentioned file?"

In case it does could someone please explain why? When I understand it I could raise a bug against SELinux to get it fixed.

Version involved:

~~~
# fwupdmgr --version
compile   org.freedesktop.fwupd         1.8.10
compile   com.hughsie.libjcat           0.1.6
runtime   org.freedesktop.fwupd-efi     1.8.10
compile   org.freedesktop.gusb          0.3.8
runtime   com.dell.libsmbios            2.4
runtime   org.freedesktop.gusb          0.3.8
runtime   org.freedesktop.fwupd         1.8.10
runtime   org.kernel                    5.14.0-284.18.1.el9_2.x86_64
~~~

Best regards,  
Tronde

**Comment by hughsie** [comment]
Weird. I guess it might be looking up the proxy information or something like that. Are you using fwupd as root? It's designed to be used as the normal user.

**Reply by Tronde** [reply]
Hi, I'm not running fwupd as root. I was root to check system settings when I encountered the log entry and run the command to get the version just once as root.

The log entry is from 03:47:10 last night. So some sort of process seem to have triggered that entry. Here is the log snipped:

~~~
Jul 03 03:47:09 example.com systemd[1]: Starting Firmware update daemon...
Jul 03 03:47:09 example.com fwupd[20837]: 01:47:09.363 FuEngine             failed to add device /sys/devices/pci0000:00/0000:00:02.1/0000:16:00.1/ata3/host2/target2:0:0/2:0:0:0/block/sr0: failed to open /dev/sr0: No medium found
Jul 03 03:47:09 example.com systemd[1]: Started Firmware update daemon.
Jul 03 03:47:09 example.com systemd[1]: Starting SETroubleshoot daemon for processing new SELinux denial logs...
Jul 03 03:47:09 example.com systemd[1]: Started SETroubleshoot daemon for processing new SELinux denial logs.
Jul 03 03:47:09 example.com systemd[1]: Started dbus-:1.1-org.fedoraproject.SetroubleshootPrivileged@2.service.
Jul 03 03:47:10 example.com systemd[1]: Starting Time & Date Service...
Jul 03 03:47:10 example.com setroubleshoot[20848]: SELinux is preventing /usr/bin/fwupdmgr from map access on the file /root/.cache/dconf/user. For complete SELinux messages run: sealert -l 41fff8c8-5296-4fdc-80ee-3276cb55142b
Jul 03 03:47:10 example.com setroubleshoot[20848]: SELinux is preventing /usr/bin/fwupdmgr from map access on the file /root/.cache/dconf/user.
                                                
                                                *****  Plugin catchall_boolean (89.3 confidence) suggests   ******************
                                                
                                                If you want to allow domain to can mmap files
                                                Then you must tell SELinux about this by enabling the 'domain_can_mmap_files' boolean.
                                                
                                                Do
                                                setsebool -P domain_can_mmap_files 1
                                                
                                                *****  Plugin catchall (11.6 confidence) suggests   **************************
                                                
                                                If you believe that fwupdmgr should be allowed map access on the user file by default.
                                                Then you should report this as a bug.
                                                You can generate a local policy module to allow this access.
                                                Do
                                                allow this access for now by executing:
                                                # ausearch -c 'fwupdagent' --raw | audit2allow -M my-fwupdagent
                                                # semodule -X 300 -i my-fwupdagent.pp
                                                
~~~

**Reply by hughsie** [reply]
out of interest, can you do fwupdmgr get-details any-firmware-file.cab as root, or do you get the same selinux denial?

**Reply by Tronde** [reply]
I get an error but no selinux event in log:

~~~
# fwupdmgr get-details any-firmware-file.cab
WARNING: UEFI capsule updates not available or enabled in firmware setup
  See https://github.com/fwupd/fwupd/wiki/PluginFlag:capsules-unsupported for more information.
failed to open any-firmware-file.cab
~~~

**Reply by hughsie** [reply]
Okay, I see this too -- seemingly when the libdconfsettings.so extension point gets started.

	newfstatat(AT_FDCWD, "/usr/lib64/gio/modules/libdconfsettings.so", {st_mode=S_IFREG|0755, st_size=62224, ...}, 0) = 0
	openat(AT_FDCWD, "/usr/lib64/gio/modules/libdconfsettings.so", O_RDONLY|O_CLOEXEC) = 6
	read(6, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
	newfstatat(6, "", {st_mode=S_IFREG|0755, st_size=62224, ...}, AT_EMPTY_PATH) = 0
	mmap(NULL, 61664, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 6, 0) = 0x7f5338b26000
	mmap(0x7f5338b2a000, 28672, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 6, 0x4000) = 0x7f5338b2a000
	mmap(0x7f5338b31000, 12288, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 6, 0xb000) = 0x7f5338b31000
	mmap(0x7f5338b34000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 6, 0xd000) = 0x7f5338b34000
	close(6)                                = 0
	mprotect(0x7f5338b34000, 4096, PROT_READ) = 0
	futex(0x7f533a7cbec8, FUTEX_WAKE_PRIVATE, 2147483647) = 0
	getuid()                                = 0
	openat(AT_FDCWD, "/run/dconf/user/0", O_RDONLY) = -1 ENOENT (No such file or directory)
	mkdir("/root/.cache", 0700)             = -1 EEXIST (File exists)
	openat(AT_FDCWD, "/root/.cache/dconf/profile", O_RDONLY) = -1 ENOENT (No such file or directory)
	openat(AT_FDCWD, "/etc/dconf/profile/user", O_RDONLY) = 6
	newfstatat(6, "", {st_mode=S_IFREG|0644, st_size=61, ...}, AT_EMPTY_PATH) = 0
	read(6, "user-db:user\nsystem-db:local\nsys"..., 4096) = 61
	read(6, "", 4096)                       = 0
	close(6)                                = 0
	futex(0x7f533a7cbec8, FUTEX_WAKE_PRIVATE, 2147483647) = 0
	futex(0x7f533a7cbec8, FUTEX_WAKE_PRIVATE, 2147483647) = 0
	mkdir("/root/.cache/dconf", 0700)       = -1 EEXIST (File exists)
	newfstatat(AT_FDCWD, "/root/.cache/dconf", {st_mode=S_IFDIR|0700, st_size=4096, ...}, 0) = 0
	openat(AT_FDCWD, "/root/.cache/dconf/user", O_RDWR|O_CREAT, 0600) = 6
	pwrite64(6, "\0", 1, 1)                 = 1
	mmap(NULL, 1, PROT_READ, MAP_SHARED, 6, 0) = 0x7f533ab87000
	close(6)                                = 0
	openat(AT_FDCWD, "/root/.config/dconf/user", O_RDONLY|O_CLOEXEC) = 6
	newfstatat(6, "", {st_mode=S_IFREG|0644, st_size=2128, ...}, AT_EMPTY_PATH) = 0
	mmap(NULL, 2128, PROT_READ, MAP_PRIVATE, 6, 0) = 0x7f533ab86000
	close(6)                                = 0

i.e. it's legitimate. Can you open a Red Hug bug against RHEL selinux-policy please.

**Comment by Tronde** [comment]
Hi, I report it as a Bug when I'm back to work tomorrow.

**Reply by Tronde** [reply]
Here is the link to the BZ for reference: [https://bugzilla.redhat.com/show_bug.cgi?id=2219542](https://bugzilla.redhat.com/show_bug.cgi?id=2219542)

**Reply by Tronde** [reply]
PR: [https://github.com/fedora-selinux/selinux-policy/pull/1774](https://github.com/fedora-selinux/selinux-policy/pull/1774)

**Reply by hughsie** [reply]
Many thanks!

**Comment by dheerapat** [comment]
Got the same warning after install virtualbox in RHEL 9.2 but not affected anything.

**Comment by tcouey** [comment]
I'm also seeing this. Not sure if it matters, but I'm also running Virtualbox in RHEL 9.2. I think I installed it around the time these started popping up too, so maybe that's what started it.

**Reply by seedneus** [reply]
I'm also running RHEL 9.2 and have this issue without installing virtualbox - just fresh install. It is resolved in newer version of selinux-policy and selinux-policy-targeted in RHEL testing repositories - but I don't want to enable anything "testing".

https://bugzilla.redhat.com/show_bug.cgi?id=2219542

**Comment by Tronde** [comment]
This has been resolved as part of [https://bugzilla.redhat.com/show_bug.cgi?id=2214581](https://bugzilla.redhat.com/show_bug.cgi?id=2214581) in errata [RHBA-2023:6617](https://access.redhat.com/errata/RHBA-2023:6617). In current RHEL 9 it does not occur anymore. Therefore I'm closing here. :-)
